src/main/java/net/mcreator/ui/workspace/WorkspacePanelVariables.java

```diff
@@ -78,12 +80,12 @@
 					if (n == JOptionPane.YES_OPTION) {
 						super.setValueAt(value, row, column);
 						if (column == 1) { // variable type has been changed
-							VariableElementType type = (VariableElementType) getValueAt(row, column);
-							if (type == VariableElementType.NUMBER) {
+							VariableElementType type = VariableElementTypeLoader.INSTANCE.getVariableTypeFromString((String) getValueAt(row, column));
+							if (type == VariableElementTypeLoader.BuiltInTypes.NUMBER) {

```

Comment:
I think this part could be edited so that it sets the variable default value instead of checking each type

Reply:
No, because ItemStack (and future variable types) use something that the majority of users won't understand. For example, if I add a Date var type, the default value will be `new Date()` and you can't really change this for a value like a number or a text. Also, someone new to MCreator or who doesn't have Java knowledge will not understand what it is. I prefer to only keep these 3 var types, so the UI is clean and understandable by everyone.